pkgs/tools/networking/pihole-ftl/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, cmake
+, fetchFromGitHub
+, gmp
+, libidn
+, nettle
+, readline
+, stdenv
+}:
+
+stdenv.mkDerivation rec {
+  pname = "pihole-ftl";
+  version = "5.8.1";
+
+  src = fetchFromGitHub {
+    owner = "pi-hole";
+    repo = "FTL";
+    rev = "v${version}";
+    sha512 = "1ybf71jwr781c7q3xvjla2vhnarba6dpp94ghzd31vhdc038g36f0ad2xp3hh8bk1hmlwzzqpcd5pl23w0f4b19kgms4cbk8rh6nvpr";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [ gmp libidn nettle readline ];
+
+  postPatch = ''
+    substituteInPlace src/CMakeLists.txt \
+      --replace "CMAKE_STATIC_LIBRARY_SUFFIX" "CMAKE_SHARED_LIBRARY_SUFFIX" \
+      --replace "EXTRAWARN \"-Werror -Waddress" "EXTRAWARN \"-Waddress"
```

Comment:
--replace "-Werror" "" is also valid and doesn't break if the surroundings change.

Reply:
I'm continuing the discussion in [this thread](https://github.com/NixOS/nixpkgs/pull/108055#discussion_r633968596)